NuRadioReco/modules/io/rno_g/readRNOGDataMattak.py

```diff
@@ -0,0 +1,644 @@
+            return False
+      
+      if self._time_high is not None:
+         time_start = astropy.time.Time(run_info["time_start"].values[0])
+         if time_start > self._time_high:
+            self.logger.info(f"Reject station {station_id} run {run_id} because run started time after {self._time_high}")
+            return False
+      
+      run_type = run_info["run_type"].values[0]
+      if not run_type in self.__run_types:
+         self.logger.info(f"Reject station {station_id} run {run_id} because of run type {run_type}")
+         return False
+      
+      trigger_rate = run_info["trigger_rate"].values[0] * units.Hz 
+      if self.__max_trigger_rate and trigger_rate > self.__max_trigger_rate:
+         self.logger.info(f"Reject station {station_id} run {run_id} because trigger rate is to high ({trigger_rate / units.Hz} Hz)")
+         return False
+      
+      return True
+
+
+   def __get_n_events_of_prev_datasets(self, dataset_idx):
+      """ Get accumulated number of events from previous datasets """
+      dataset_idx_prev = dataset_idx - 1
+      return int(self._event_idxs_datasets[dataset_idx_prev]) if dataset_idx_prev >= 0 else 0
+      
+   
+   def __get_dataset_for_event(self, event_idx):
+      """ Get correct dataset and set entry accordingly to event index
+      
+      Parameters
+      ----------
+      
+      event_index: int
+         Same as in read_event().
+      
+      Returns
+      -------
+      
+      dataset: mattak.Dataset.Dataset
+      """
+      # find correct dataset
+      dataset_idx = np.digitize(event_idx, self._event_idxs_datasets)
+      dataset = self._datasets[dataset_idx]
+
+      event_idx_in_dataset = event_idx - self.__get_n_events_of_prev_datasets(dataset_idx)
+      dataset.setEntries(event_idx_in_dataset)  # increment iterator -> point to new event
+            
+      return dataset
+   
+   
+   def filter_event(self, evtinfo, event_idx=None):
+      """ Filter an event base on its EventInfo and the configured selectors.
+
+      Parameters
+      ----------
+      
+      event_info: mattak.Dataset.EventInfo
+         The event info object for one event.
+      
+      event_index: int
+         Same as in read_event(). Only use for logger.info(). (Default: None)
+         
+      Returns
+      -------
+      
+      skip: bool
+         Returns True to skip/reject event, return False to keep/read event
+      """
+      if self._selectors is not None:
+         for selector in self._selectors:
+            if not selector(evtinfo):
+               self.logger.debug(f"Event {event_idx} (station {evtinfo.station}, run {evtinfo.run}, "
+                                 f"event number {evtinfo.eventNumber}) is skipped.")
+               self.__skipped += 1
+               return True
+      
+      return False
+   
+   
+   def get_event_informations(self, keys=["station", "run", "eventNumber"]):
+      """ Return information of all events from the EventInfo
+      
```

Comment:
pedantic, but maybe remove the s at the end (information is uncountable in English)

Reply:
I know.. the proper name would be `get_events_information` but that sounded bad to me.